awx/main/tasks/facts.py

```diff
@@ -88,17 +89,17 @@ def update_hosts(host_list, max_tries=5):
     msg='Inventory {inventory_id} host facts: updated {updated_ct}, cleared {cleared_ct}, unchanged {unmodified_ct}, took {delta:.3f} s',
     add_log_data=True,
 )
-def finish_fact_cache(hosts, destination, facts_write_time, log_data, job_id=None, inventory_id=None):
+def finish_fact_cache(hosts_cached, destination, facts_write_time, log_data, job_id=None, inventory_id=None):
     log_data['inventory_id'] = inventory_id
     log_data['updated_ct'] = 0
     log_data['unmodified_ct'] = 0
     log_data['cleared_ct'] = 0
 
-    if isinstance(hosts, QuerySet):
-        hosts = hosts.iterator()
+    if isinstance(hosts_cached, QuerySet):
+        hosts_cached = hosts_cached.iterator()
```

Comment:
this `if` should be deleted, not modified. There is only 1 call we care about. With your changes, it only takes in what the `start_fact_cache` returns, which is a `list()` on creation and there's nothing that can ever change that later.